core/thread_flags.c

```diff
@@ -80,7 +80,8 @@ thread_flags_t thread_flags_wait_one(thread_flags_t mask)
     _thread_flags_wait_any(mask);
     thread_t *me = (thread_t*) sched_active_thread;
     unsigned tmp = me->flags & mask;
-    return _thread_flags_clear_atomic(me, thread_flags_clear(1 << bitarithm_lsb(tmp)));
+    tmp &= -tmp;
```

Comment:
hmm sounds sensible. If someone wants to compile for a PDP, then it might choose one's complement instead of 2's